p2panda-net/src/engine/engine.rs

```diff
@@ -545,13 +562,9 @@ impl EngineActor {
         self.join_earmarked_topics().await?;
 
         // Inform the connection manager about the peer topics
-        //
-        // NOTE: This will only return once sync has been attempted with all novel peer-topic
-        // combinations.
-        if let Some(ref mut sync_manager) = self.sync_manager {
-            sync_manager
-                .update_peer_topics(delivered_from, topics)
-                .await?;
+        if let Some(sync_manager_tx) = &self.sync_manager_tx {
+            let peer_topics = ToSyncManager::new(delivered_from, topics);
+            sync_manager_tx.send(peer_topics).await?
```

Comment:
👍🏻

Reply:
Good call on avoiding having to debug unexpected blocking behaviour in the future! I've added a capacity check: if the queue is full then we call `send_timeout()` which will only block for the specified duration before returning an error. Are you happy with this behaviour for now? I've set the timeout to 100 ms (feels quite generous).